### ISSUE TITLE
some grafana dashboard fields are empty

### ISSUE THREAD

**Issue by arifroni** [ar51an/unbound-dashboard#11]
Hello

I followed the instruction and i had to make some changes. I am using unbound with adguard in a proxmox LXC.
1. i used amd64 binary from [here](https://github.com/ar51an/unbound-exporter/tree/main/bin/amd64)
2. i changed the prometheus-unbound-exporter.service like this
![image](https://github.com/user-attachments/assets/c47b66e2-2b3c-4abd-a0ec-defcd043e76a)

But i dont get all the fields. Since i am not using any blocklists, i expected some empty box, but all.
![image](https://github.com/user-attachments/assets/1f5e7e0d-c200-4198-b389-4126e21019f0)

Any idea what could be the issue?

**Comment by ar51an** [comment]
1) The unbound-exporter is working fine, that is why you are seeing the stats in your screenshot. If you are not using blocklist, ~only panel that would be empty is the first panel in the second row~. If you want, you can remove it and resize/adjust subsequent panles in second row. (as shown below)

![unbound-1](https://github.com/user-attachments/assets/017337c3-9e42-4309-96df-a7e1cb027856)

![unbound-2](https://github.com/user-attachments/assets/3b6e1886-e785-40b0-a2dd-a01d918eaa10)

![unbound-3](https://github.com/user-attachments/assets/67beb349-97c9-4b75-84e8-12baefa07c48)

2) All other panels that are not showing any data in your screenshot are populated through loki and unbound log.
Did you install the Loki and Promtail?
Did you enable the unbound logging?
As mentioned under the Loki section in the instructions, since you are on amd64 so get amd64 deb files for loki and promtail from below links:
https://github.com/grafana/loki/releases/download/v2.8.4/loki_2.8.4_amd64.deb
https://github.com/grafana/loki/releases/download/v2.8.4/promtail_2.8.4_amd64.deb
What is the output of below cmds:
`sudo systemctl status loki`
`sudo systemctl status promtail`
`ls -l /var/log/unbound/`

**Comment by arifroni** [comment]
thanks for your reply. I have installed right deb file, i think :)
unbound.conf
![image](https://github.com/user-attachments/assets/b950c036-ec40-4c49-acb4-e4d3bfb457f2)

here are the outputs:
![image](https://github.com/user-attachments/assets/fd5da530-c40b-4c55-8801-8d75744514cb)
![image](https://github.com/user-attachments/assets/dfeeab05-5b3e-4fbb-82f2-8c72a25c9f68)

could it be that the log file has permission issue? thats why unbound cant export log file to that directory?

**Comment by ar51an** [comment]
Seems like you are running the distro version of Unbound, it runs as chroot by apparmor.

Modify apparmor policy profile for unbound:
`sudo nano \etc\apparmor.d\usr.sbin.unbound`
(Location of usr.sbin.unbound may vary depending on your distro)

Add below line at the end of the file before closing curly bracket:
`/var/log/unbound/unbound.log rw,`

Reload apparmor policy profile for unbound:
`sudo apparmor_parser -r /etc/apparmor.d/usr.sbin.unbound`
(Provide the correct location of the policy profile that you used in the first step)

Restart unbound / Reboot machine, verify that /var/log/unbound/ dir contains unbound.log

**Comment by arifroni** [comment]
![image](https://github.com/user-attachments/assets/0c523440-f1a5-41af-a465-107de9af5d27)
![image](https://github.com/user-attachments/assets/55a62583-4201-4aaf-8ba3-eaf55ad5966f)
Any idea what could be the problem here?

**Comment by ar51an** [comment]
Try below:

`sudo nano /etc/apparmor.d/local/usr.sbin.unbound`
(It will create the file if it does not exist)

Add below line:
`/var/log/unbound/unbound.log rw,`

`sudo apparmor_parser -r /etc/apparmor.d/usr.sbin.unbound`
(It should apply the changes from `/etc/apparmor.d/local/usr.sbin.unbound` to main profile `/etc/apparmor.d/usr.sbin.unbound`)

**Comment by ar51an** [comment]
Closing due to no activity.

**Comment by arifroni** [comment]
Hi, i couldnt check in the last days.
but still i have same error.
![image](https://github.com/user-attachments/assets/113c8d48-29a3-4cdf-b12f-e18e74b17fca)
![image](https://github.com/user-attachments/assets/8367e250-f6ba-46b9-b739-0587002adcda)

**Comment by ar51an** [comment]
Seems like you are running a privileged lxc container.

Try to change the apparmor profile to unconfined. Then run the unbound command mentioned in the previous message.
* Add to the .conf file for container:
  > `lxc.apparmor.profile:unconfined`
  > `lxc.cap.drop:`

If above does not work then you may need to change it to unconfined in the compose file.
```
security_opt:
- apparmor:unconfined
```

**Comment by arifroni** [comment]
after adding the lines in lxc conf file, apparmor_parser -r /etc/apparmor.d/usr.sbin.unbound works without any error. but unbound is not starting anymore.
![image](https://github.com/user-attachments/assets/5389d152-c94c-4240-9262-8d52486ed079)
![image](https://github.com/user-attachments/assets/b6a73e3a-4c97-48d4-9fb4-263653ff10bb)

**Comment by arifroni** [comment]
update: after deleting 2 lines from conf file of the lxc, unbound starts again. i created the unbound.log file manually and since then i can see some of the fields are populated. except marked boxes. 
![image](https://github.com/user-attachments/assets/a720bddb-3939-4de4-ae8e-8ba5129df177)

**Comment by ar51an** [comment]
Are the bottom 4 panels (that shows domains requested) populated properly?

If yes, give it half an hour, there is not enough data for grafana yet.

**Comment by ar51an** [comment]
> after adding the lines in lxc conf file, apparmor_parser -r /etc/apparmor.d/usr.sbin.unbound works without any error. but unbound is not starting anymore. ![image](https://private-user-images.githubusercontent.com/59896374/352997629-5389d152-c94c-4240-9262-8d52486ed079.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyNzUzOTIsIm5iZiI6MTcyMjI3NTA5MiwicGF0aCI6Ii81OTg5NjM3NC8zNTI5OTc2MjktNTM4OWQxNTItYzk0Yy00MjQwLTkyNjItOGQ1MjQ4NmVkMDc5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDE3NDQ1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIwYzQ5M2IyMjA5MWViODVkM2M1YTc4MGM5YTNjYzg5MDNjMDRjNDljM2E1Zjk2YTNlOGNjOWI2OWJjZTU2YjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9fYuZGcd4FXR9PvZsITENGwPi2kz5TWnh8qMt_Ug6YQ) ![image](https://private-user-images.githubusercontent.com/59896374/352997731-b6a73e3a-4c97-48d4-9fb4-263653ff10bb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyNzUzOTIsIm5iZiI6MTcyMjI3NTA5MiwicGF0aCI6Ii81OTg5NjM3NC8zNTI5OTc3MzEtYjZhNzNlM2EtNGM5Ny00OGQ0LTlmYjQtMjYzNjUzZmYxMGJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDE3NDQ1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMwNjUyYThlODRjODQ3ZTZlOTU1ZjVjMmU3Zjc3ZTRhMTgxOGU2M2VmY2IzMjUwZjBkMTg1NzliZTQ1MzQwNzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Nwrd5XxJqqJo6DGBBkOeXfeJLSMWOpYsg8tShnW07l0)

I know you reverted the changes. Just mentioning, it may be helpful to you in future.
The above error in the screenshot tells nothing it is just the last status. Actual error happened before that. You should find the actual error to figure out what went wrong with below cmd:
`sudo journalctl -u unbound -n 200 -f`

**Comment by ar51an** [comment]
> update: after deleting 2 lines from conf file of the lxc, unbound starts again. i created the unbound.log file manually and since then i can see some of the fields are populated. except marked boxes. ![image](https://private-user-images.githubusercontent.com/59896374/353119556-a720bddb-3939-4de4-ae8e-8ba5129df177.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyNzY2MDYsIm5iZiI6MTcyMjI3NjMwNiwicGF0aCI6Ii81OTg5NjM3NC8zNTMxMTk1NTYtYTcyMGJkZGItMzkzOS00ZGU0LWFlOGUtOGJhNTEyOWRmMTc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDE4MDUwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5ODQyZDVkNGJkNjRiYjc3YTg4YTI0Y2Q3ZTQ4ZDhkYTJlZDQ1MmRkM2EwNDIyZjlhN2Y2OTEzZTJmYTQ1ZDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pmBPskBIne_H3Pnl9NdIf0VvzgZwPSaz_jmTvmebvDw)

Are the bottom 4 panels (that shows domains requested) populated properly?
Give it half an hour to have enough data for grafana. These two panels are linked with the unbound blocked count, since you are not using the unbound blocklist you might not see anything in these two panels.

![image](https://github.com/user-attachments/assets/50d0751f-8bf7-4e80-b917-a8e102ba06a3)

**Comment by arifroni** [comment]
Thanks a lot for all your help :). I will take a look on the error.
As for the 2 missing box (Queries & Queries), it make sense that i wont see them. I didnt realize it shows blocked website data.
Domain requests panels show data, but 2 not 4. I guess missing panels are related to blocked domains. clients are always 127.0.0.1, because adguard is requesting, right?

![image](https://github.com/user-attachments/assets/36a5fe43-cad8-4bfa-aa21-77b56a67e902)

**Comment by ar51an** [comment]
Yes right two bottom panels are also populated with blocked domains (did not realize in my first comment). Check the dashboard v2.2 screenshot at:
https://github.com/ar51an/unbound-dashboard/blob/main/screenshots/screenshots.md

> clients are always 127.0.0.1, because adguard is requesting, right?

Check the output of:
`cat /etc/resolv.conf`
It should have the DNS server IP as `nameserver <IP>` only.

OR

`dig google.com`
Check the third last line (could vary on your system) that starts with `SERVER: <IP>`. IP should be your DNS server IP.

What are the Client IPs it shows in Adguard logs? (Actual client IPs OR 127.0.0.1)

Bottom line is either Adgurad is not forwarding client IPs to unbound (should be some setting or configuration to enable it) OR you need to set DNS server IP under LAN DNS server on your router.

**Comment by arifroni** [comment]
output:
![image](https://github.com/user-attachments/assets/437173e0-74ee-4cdb-b174-e2babee37cf4)
i can see local ips in adguard.
![image](https://github.com/user-attachments/assets/1c4b7ce4-f4cb-48a5-bc4b-33f2589e5227)
I found this option in adguard, that maybe has something to do with ip forwarding (i am far from expert on this topic as you can imagine :D)
![image](https://github.com/user-attachments/assets/1fe4c440-030a-4802-a79d-0d16e589a036)
But turning it on makes no difference.
